akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -0,0 +1,247 @@
+   * given name.
+   */
+  def stop(childName: String): Unit
+
+  /**
+   * Register for [[Terminated]] notification once the Actor identified by the
+   * given [[ActorRef]] terminates. This notification is also generated when the
+   * [[ActorSystem]] to which the referenced Actor belongs is declared as
+   * failed (e.g. in reaction to being unreachable).
+   */
+  def watch[U](other: ActorRef[U]): ActorRef[U]
+
+  /**
+   * Register for [[Terminated]] notification once the Actor identified by the
+   * given [[akka.actor.ActorRef]] terminates. This notification is also generated when the
+   * [[ActorSystem]] to which the referenced Actor belongs is declared as
+   * failed (e.g. in reaction to being unreachable).
+   */
+  def watch(other: akka.actor.ActorRef): akka.actor.ActorRef
+
+  /**
+   * Revoke the registration established by [[#watch[U]* watch]]. A [[Terminated]]
+   * notification will not subsequently be received for the referenced Actor.
+   */
+  def unwatch[U](other: ActorRef[U]): ActorRef[U]
+
+  /**
+   * Revoke the registration established by [[#watch(* watch]]. A [[Terminated]]
+   * notification will not subsequently be received for the referenced Actor.
+   */
+  def unwatch(other: akka.actor.ActorRef): akka.actor.ActorRef
+
+  /**
+   * Schedule the sending of a [[ReceiveTimeout]] notification in case no other
+   * message is received during the given period of time. The timeout starts anew
+   * with each received message. Provide `Duration.Undefined` to switch off this
+   * mechanism.
+   */
+  def setReceiveTimeout(d: Duration): Unit
+
+  /**
+   * When handling a [[Failed]] signal the behavior should invoke this method to
+   * inform the execution mechanism of the supervisor decision. If this is not
+   * done then the implied decision is to escalate the failure.
+   */
+  def setFailureResponse(decision: Failed.Decision): Unit
+
+  /**
+   * Schedule the sending of the given message to the given target Actor after
+   * the given time period has elapsed. The scheduled action can be cancelled
+   * by invoking [[akka.actor.Cancellable!.cancel* cancel]] on the returned
+   * handle.
+   */
+  def schedule[U](delay: FiniteDuration, target: ActorRef[U], msg: U): a.Cancellable
+
+  /**
+   * This Actor’s execution context. It can be used to run asynchronous tasks
+   * like [[scala.concurrent.Future]] combinators.
+   */
+  implicit def executionContext: ExecutionContextExecutor
+
+  /**
+   * Create a child actor that will wrap messages such that other Actor’s
+   * protocols can be ingested by this Actor. You are strongly advised to cache
+   * these ActorRefs or to stop them when no longer needed.
+   */
+  def createWrapper[U](f: U ⇒ T): ActorRef[U]
+}
+
+/**
+ * An [[ActorContext]] for testing purposes that provides only stubs for the
+ * effects an Actor can perform and replaces created child Actors by
+ * [[Inbox$.sync* a synchronous Inbox]].
+ */
+class DummyActorContext[T](
```

Comment:
so far I don’t think so: Behavior is constructed such that it can be run in different contexts, so providing those contexts seems natural